File: MovieFinder.Core/Dtos/PaginatedDto.cs
namespace MovieFinder.Core.Dtos;

public class PaginatedDto<T> where T : class
{
    public PaginatedDto(int pageIndex, int pageSize, int count, IEnumerable<T> data)
    {
        PaginationMeta = new PaginationMetaDto(pageIndex, pageSize, count);
        Data = data;
    }

    public PaginationMetaDto PaginationMeta { get; set; }
    public IEnumerable<T> Data { get; set; }
}

File: MovieFinder.Web/ClientApp/src/core/models/PaginatedDto.ts
import {PaginationMetaDto} from "./paginationMetaDto";

export class PageinatedDto<T>{
    data: T;
    paginationMeta: PaginationMetaDto

    constructor(data: T, paginationMeta : PaginationMetaDto) {
        this.data = data;
        this.paginationMeta = paginationMeta;
    }
}